pkg/volume/configmap/configmap.go

```diff
@@ -202,22 +202,36 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	return nil
 }
 
-func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string][]byte, error) {
-	payload := make(map[string][]byte, len(configMap.Data))
+func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap, defaultMode *int32) (map[string]volumeutil.FileProjection, error) {
+	if defaultMode == nil {
+		return nil, fmt.Errorf("No defaultMode used, not even the default value for it")
```

Comment:
I'm not sure about `invalid default mode...` because validation is done at another layer. This is just a check to make sure we don't dereference a null pointer and "fail better".

It may affect mostly tests (unit or e2e), because the struct can be created manually in a test and might not have this field.

@pmorie What about `default mode not specified and is not optional` or something along those lines? What do you think?